### PR TITLE
Update tools in CI

### PR DESCRIPTION
**Pull request by hasufell**

### CODE REVIEW

**Comment by hasufell** [comment]
kinda blocked by https://github.com/haskell/ghcup-hs/issues/962

**Comment by runeksvendsen** [comment]
@hasufell what's the motivation behind this PR?

**Comment by runeksvendsen** [comment]
> kinda blocked by #962

Given that GHC 9.4.8 is broken in many different ways (cf. https://github.com/haskell/ghcup-hs/issues/969) I think it would make sense to not use it in the CI pipeline.

**Comment by hasufell** [comment]
https://github.com/haskell/ghcup-hs/issues/969

We should always build ghcup with the recommended version. Even before that GHC is recommended.

**Comment by hasufell** [comment]
> > kinda blocked by #962
> 
> Given that GHC 9.4.8 is broken in many different ways (cf. https://github.com/haskell/ghcup-hs/issues/969) I think it would make sense to not use it in the CI pipeline.

The only real GHC bug might be the alpine segfault. Everything else are ecosystem issues. And for that I have not submitted a bug report. I don't have a small reproducer.

**Comment by runeksvendsen** [comment]
> #969
> 
> We should always build ghcup with the recommended version. Even before that GHC is recommended.

Would it be viable to do this _as well as_ building with other, known-to-work versions of GHC? Instead of just picking a single GHC version?

If there are known issues building stuff with GHC 9.4.8, but we also build with e.g. GHC 9.4.5 in CI and this succeeds, then I see nothing wrong with merging the PR, as it's clear from the CI runs that it's GHC version 9.4.8 that's problematic (and we have issues to track this).

**Comment by hasufell** [comment]
> Agreed. Would it be viable to also build with other, known-to-work versions of GHC? Instead of picking a single GHC version?

Well, right now we don't make much of a distinction between PR build and release build.

I don't want more than one GHC build in the release pipeline. So that would require splitting the workflows.

File: ghcup.cabal [hunk 1]
```diff
@@ -327,12 +327,10 @@ executable ghcup
       , brick         ^>=2.1
       , transformers  ^>=0.5
       , vty           ^>=6.0
-      , unix          ^>=2.7
```
[the remark this answers is not in the document]
Comment (hasufell) [reply ghcup.cabal:330]: It fixes windows CI as the commit says

[the remark this answers is not in the document]
Comment (hasufell) [reply ghcup.cabal:330]: I don't split my work into 3 or 4 different PRs when I'm actively working on CI in general. It's just confusing. 

So we'll have to make a decision here what to do with i386 alpine and move the PR forward

**Comment by runeksvendsen** [comment]
> I don't want more than one GHC build in the release pipeline.

What's the reason for this? Slowness?

**Comment by hasufell** [comment]
> > I don't want more than one GHC build in the release pipeline.
> 
> What's the reason for this? Slowness? 

The release pipeline is there to ship the ghcup binaries. We want as few points of failure as possible.

**Pull request merged**